src/poetry/core/factory.py

```diff
@@ -2,10 +2,10 @@
 
 import logging
 
+from collections.abc import Mapping
```

Comment:
Shouldn't this be in type checking?

Reply:
No, because it's used in an instance check.